File: java/leetcode_java/src/RichestCustomerWealth1672.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class RichestCustomerWealth1672 {
    public int calculateRichestCustomerWealth(List<List<Integer>> accounts) {
        //System.out.print("1672. Richest Customer Wealth");
        int highest_balance = 0;

        System.out.println("Accounts:");
        // sum each account and see who has the highest balance
        for (List<Integer> row : accounts) {
            System.out.println(row);
            int sum = 0;
            for (int num : row) {
                sum += num;
            }
            highest_balance = Math.max(sum, highest_balance);
        }

        return highest_balance;
    }
}

File: javascript/leetcode_js/src/richest_customer_wealth_1672.js
// richest_customer_wealth_1672.js
// 1672. Richest Customer Wealth
// sum each account
// return highest balance

export function calculate_richest_customer_wealth(accounts) {
    let highest_balance = 0;

    // sum each account and see who has the highest balance
    for (let row of accounts) {
        let sum = row.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        console.log("sum: ", sum)
        highest_balance = (sum > highest_balance) ? sum : highest_balance;
    }

    return highest_balance
}

